Program.cs

```diff
@@ -3,46 +3,61 @@
-            var destination = System.Environment.GetEnvironmentVariable("dropdesitnation")  ?? "/drop";
-            var url = System.Environment.GetEnvironmentVariable("dropurl") 
+            var relativePath = System.Environment.GetEnvironmentVariable(RelavePathEnvironmentVariable);
+            var pat = System.Environment.GetEnvironmentVariable(VSTSPatEnvironmentVariable);
+            var destination = System.Environment.GetEnvironmentVariable(DropDestinationEnvironmentVariable)
+                              ?? DefaultDropDestination;
+            var url = System.Environment.GetEnvironmentVariable(DropUrlEnvironmentVariable)
                       ?? ExtractDropUrl(destination);
 
+            // sample URL:
             // https://msasg.artifacts.visualstudio.com/DefaultCollection/_apis/drop/drops/Aether_master/7dd31c59986465bfa9af3bd883cb35ce132979a2/e90d7f94-265a-86c7-5958-66983fdcaa06
-            Console.WriteLine($"url:{url}");
+            logger.LogInformation($"url: {url}");
             // /Release/Amd64/app/aether/AetherBackend
-            Console.WriteLine($"relativepath:{relativepath}");
-            //not good security practice to write out your PAT
-            //Console.WriteLine($"pat:{pat}");
-            Console.WriteLine($"dest:{destination}");
+            logger.LogInformation($"relative path: {relativePath}");
+            logger.LogInformation($"destination: {destination}");
+            var proxy = new VSTSDropProxy(url, relativePath, pat, loggerFactory);
             var sw = Stopwatch.StartNew();
-            //should be evironment variable SYSTEM_ACCESSTOKEN
-            var proxy = new VSTSDropProxy(url, relativepath, pat);
             proxy.Materialize(destination).Wait();
-            Console.WriteLine($"Finished in {sw.Elapsed} ");
+            sw.Stop();
+
+            logger.LogInformation($"Finished in {sw.Elapsed}");
         }
 
-        //agent based tasks automatically download artifacts from the build. 
-        //when the build only produces a vsts drop that artifact is a single json
-        //it resides in <builddefname>/<guid>/VSTSDrop.json
-        private static string ExtractDropUrl(string workingdir)
+        // agent based tasks automatically download artifacts from the build. 
+        // when the build only produces a vsts drop that artifact is a single json
+        // it resides in <builddefname>/<guid>/VSTSDrop.json
+        private static string ExtractDropUrl(string workingDirectory)
         {
             //could take an envdir on what the build dir is for now though we just have one build
-            var builddir = Directory.GetDirectories(workingdir).Single();
-            var guiddir = Directory.GetDirectories(builddir).Single();
-            var json = Path.Combine(workingdir, builddir,  guiddir, "VSTSDrop.json");
+            var buildDirectory = Directory.GetDirectories(workingDirectory).Single();
```

Comment:
poor ori just hit this sadness. Need to log what we do se when single fails oh well later change.

Reply:
TODO in future